polkadot-parachains/src/service.rs

```diff
@@ -316,51 +293,30 @@ where
 		hwbench.clone(),
 	)
 	.await
-	.map_err(|e| match e {
-		RelayChainError::Application(x) => x,
-		s => s.to_string().into(),
-	})?;
-
-	let block_announce_validator = BlockAnnounceValidator::new(relay_chain_interface.clone(), id);
+	.map_err(|e| sc_service::Error::Application(Box::new(e) as Box<_>))?;
 
 	let force_authoring = parachain_config.force_authoring;
 	let validator = parachain_config.role.is_authority();
 	let prometheus_registry = parachain_config.prometheus_registry().cloned();
 	let transaction_pool = params.transaction_pool.clone();
 	let import_queue_service = params.import_queue.service();
+	let net_config = FullNetworkConfiguration::new(&parachain_config.network);
 
 	let (network, system_rpc_tx, tx_handler_controller, start_network, sync_service) =
-		sc_service::build_network(sc_service::BuildNetworkParams {
-			config: &parachain_config,
+		build_network(BuildNetworkParams {
+			parachain_config: &parachain_config,
+			net_config,
 			client: client.clone(),
 			transaction_pool: transaction_pool.clone(),
+			para_id,
 			spawn_handle: task_manager.spawn_handle(),
+			relay_chain_interface: relay_chain_interface.clone(),
 			import_queue: params.import_queue,
-			block_announce_validator_builder: Some(Box::new(|_| {
-				Box::new(block_announce_validator)
-			})),
-			warp_sync_params: None,
-		})?;
-
-	let rpc_builder = {
-		let client = client.clone();
-		let transaction_pool = transaction_pool.clone();
-		let backend = backend.clone();
-		let offchain_indexing_enabled = parachain_config.offchain_worker.indexing_enabled;
-
-		// `backend` and offchain_indexing_enabled` are encointer customizations.
-		Box::new(move |deny_unsafe, _| {
-			let deps = rpc::FullDeps {
-				client: client.clone(),
-				pool: transaction_pool.clone(),
-				backend: backend.clone(),
-				offchain_indexing_enabled,
-				deny_unsafe,
-			};
-
-			rpc_extensions(deps)
 		})
-	};
```

Comment:
Have you tried running this? we made here some adjustments for our custom RPCs, it seems weird to me now that we remove this.

Reply:
True, good catch